pkg/webhook/controlplane/genericmutator/mutator.go

```diff
@@ -48,29 +48,38 @@ type EnsurerContext interface {
 // If they don't initially, they are mutated accordingly.
 type Ensurer interface {
 	// EnsureKubeAPIServerService ensures that the kube-apiserver service conforms to the provider requirements.
-	EnsureKubeAPIServerService(context.Context, EnsurerContext, *corev1.Service) error
+	// "old" might be "nil" and must always be checked.
+	EnsureKubeAPIServerService(ctx context.Context, etcx EnsurerContext, new, old *corev1.Service) error
```

Comment:
@ialidzhikov's comment https://github.com/gardener/gardener-extensions/pull/622#discussion_r394410496 is not addressed here yet